text/0207-context-scoped-attributes.md

```diff
@@ -0,0 +1,278 @@
+
+Context-scoped attributes should be thought of equivalent to adding the attribute directly to each single telemetry item it applies to.
+
+Context-scoped attributes MUST NOT be propagated cross-service, no context propagator should be implemented for them. This is because they are meant to annotate (a subset of) the spans of a single service (see also [the next section](#comp-baggage)).
+
+<a name="comp-baggage"></a>
+
+### Comparing Context-scoped attributes to Baggage
+
+Context-scoped attributes and [baggage](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md) have two major commonalities:
+
+* Both are "bags of attributes"
+* Both are propagated identically in-process (Baggage also lives on the Context).
+
+However, the use cases are very different: Baggage is meant to transport app-level data along the path of execution, while Context-scoped attributes are annotations for telemetry in the same execution path.
+
+This also explains the functional & API differences:
+
+* Most important point: Baggage is meant to be propagated; in fact, this is the main use case of baggage. While a propagator for Context-scoped attributes would be implementable, it would break the intended meaning by extending the scope of attributes to called services and would also raise many security concerns.
+* Baggage is both readable and writable for the application, while Context-scoped attributes, like all other telemetry attributes, are write-only.
+* Baggage only supports string values.
+* Baggage is not available to telemetry exporters (although e.g., a SpanProcessor could be used to change that), while that's the whole purpose of Context-scoped attributes.
+
+### Comparing Context-scoped attributes to Instrumentation Scope Attributes
+
+Context-scoped attributes and the recently added [Instrumentation Scope attributes](https://github.com/open-telemetry/oteps/pull/201) have these commonalities:
+
+* Both have "scope" in the name
+* Both apply a set of telemetry attributes to a set of telemetry items (those in their "scope")
+
+While Instrumentation Scope attributes apply to all items emitted by the same Tracer, Meter or LogEmitter (i.e., typically the same telemetry-implementing unit of code),
+the scope of Context-scoped attributes is determined at runtime by the Context.
+This also means that the values for instrumentation scope are usually defined at build time,
+while context-scoped attributes will most often have values determined at runtime.
+
+In practice, Instrumentation scope and Context scope will [often be completely orthogonal](#scope-and-context):
+When the Context is flows in-process through a single
+service, there will often be exactly one span per Tracer (e.g., one span from the HTTP server instrumentation as the request arrives,
+and another one from a HTTP client instrumentation, as a downstream service is called)
+
+## Internal details
```

Comment:
This needs to address concerns between Metrics + Logs/Traces.  Effectively for metrics you have two factors:

1. Cardinality issues with automatically include high-cardinality contextual attributes.
2. Views where users can configure a default set of attributes, removing high-cardinality ones.

As such, you may actually need Context attributes to include some kind of tag (default true) for whether or not they interact with Metric Signals.

Reply:
This seems like a valid concern, and configurability (which this OTEP so far lacks completely) has been brought up in comments by others several times already. 

Maybe this OTEP is actually too simple and we need something more complicated to be actually useful in practice. As for how to do this, I don't think it makes sense to let the instrumentation decide this, since it will probably depend mostly on what backend the user is using for metrics, maybe even what kind of license they have or how much they are willing to pay and how important a particular attribute is for a particular metric and such things.

So I think we'd need a way to let the user configure, as part of their SDK configuration, which context-scoped attributes they want to apply per-metric. This could of course be extended to other signals on a granularity level that makes sense (e.g. per span name or per tracer).

I suggested an extremely minimal "configuration" I could add in https://github.com/open-telemetry/oteps/pull/207#issuecomment-1242175278, namely a simple per-signal on-off switch that the user would set e.g. when creating the MeterProvider to indicate whether they want context attributes for that signal.

My suggestion would still be to add configurability as an add-on after this OTEP. If you think that this is not good enough, I can try to amend the OTEP with specification for configurability. Especially for logs and metrics, I would appreciate help with how this could look. For the latter, would Views be the way to go?